clerk/authorization.go

```diff
@@ -0,0 +1,47 @@
+package clerk
+
+import "slices"
+
+type CheckAuthorizationParams struct {
+	Permission string
+	Role       string
+}
+
+// CheckAuthorization verifies if the user has the given permission or role.
+// Performing role checks is not considered a best-practice and
+// developers should avoid it as much as possible.
+// Usually, complex role checks can be refactored with a single permission check.
+func (c *client) CheckAuthorization(token string, params CheckAuthorizationParams) (bool, error) {
```

Comment:
❓ I'm not sure if the design has been already discussed and analyzed in depth but is the current API preferred over explicit methods such as:

```go
func (c *client) TokenHasPermission(token string, permission string) (bool, error) {}
func (c *client) TokenHasRole(token string, role string) (bool, error) {}
```

Perhaps we try to have parity with the JS SDK methods and I'm unaware of it. Asking this now, because it will be difficult to deprecate this helper afterwards!

Reply:
I agree with George on that, i believe the `CheckAuthorization()` helper doesn't make sense at this point. As a developer i would expect to check if a user has the proper role/permission using the sessionClaims, which are already populated and validated from the token and available on my Backend. I believe the methods on the sessionClaims are good enough to enforce elevated access control